hail/src/main/scala/is/hail/io/fs/TerraAzureStorageFS.scala

```diff
@@ -77,6 +88,16 @@ class TerraAzureStorageFS extends AzureStorageFS() {
     req.setURI(uri)
 
     val sasToken = using(httpClient.execute(req)) { resp =>
+      val statusCode = resp.getStatusLine.getStatusCode
+      if (statusCode < 200 || statusCode >= 300) {
+        val entity = resp.getEntity
+        val message =
+          if (entity != null)
+            EntityUtils.toString(entity)
+          else
+            null
+        throw new ClientResponseException(statusCode, message)
```

Comment:
Terra getSasToken was occasionally throwing 502 errors at high load -- this causes Hail to retry.